### ISSUE TITLE
With typescript-eslint, no-vars incorrectly fails for `declare var`

### ISSUE THREAD

**Issue by vhfmag** [eslint/eslint#11358]
**Tell us about your environment**

* **ESLint Version:** 5.13.0
* **Node Version:** 8.14.0
* **npm Version:** 6.6.0

<details>
<summary>Other relevant packages' depedencies</summary>
@typescript-eslint/eslint-plugin: 1.2.0
@typescript-eslint/parser: 1.2.0
</details>

**What parser (default, Babel-ESLint, etc.) are you using?**
@typescript-eslint/typescript-estree

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "extends": ["airbnb", "plugin:prettier/recommended", "prettier/react"],
  "parser": "@typescript-eslint/parser",
  "plugins": ["@typescript-eslint", "jsx-control-statements", "prettier"],
  "globals": {
    "fetch": true,
    "window": true,
    "document": true,
    "CustomEvent": true,
    "localStorage": true,
    "sessionStorage": true,
    "Notification": true,
    "If": true,
    "Else": true,
    "For": true,
    "Choose": true,
    "When": true,
    "Otherwise": true
  },
  "rules": {
    "no-unused-vars": 0,
    "prettier/prettier": "error",
    "class-methods-use-this": 0,
    "comma-dangle": [
      "error",
      {
        "arrays": "always-multiline",
        "objects": "always-multiline",
        "imports": "always-multiline",
        "exports": "always-multiline",
        "functions": "ignore"
      }
    ],
    "arrow-parens": [2, "as-needed"],
    "global-require": 0,
    "import/no-extraneous-dependencies": [
      "error",
      {
        "devDependencies": ["**/tests/*", "**/test/*"]
      }
    ],
    "import/no-named-as-default": 0,
    "import/no-unresolved": 0,
    "import/prefer-default-export": 0,
    "import/extensions": [
      2,
      {
        "js": "never",
        "jsx": "never",
        "graphql": "always"
      }
    ],

    "jsx-a11y/anchor-is-valid": [
      "error",
      {
        "components": ["Link"],
        "specialLink": ["to"]
      }
    ],
    "jsx-a11y/label-has-for": 0,
    "jsx-a11y/click-events-have-key-events": 0,
    "jsx-a11y/no-static-element-interactions": 0,
    "jsx-control-statements/jsx-for-require-each": 2,
    "jsx-control-statements/jsx-jcs-no-undef": 2,
    "no-console": [2, { "allow": ["warn", "error"] }],
    "no-duplicate-imports": 0,
    "no-undef": 0,
    "no-underscore-dangle": 0,
    "react/jsx-indent": 0,
    "react/jsx-no-undef": [2, { "allowGlobals": true }],
    "react/jsx-filename-extension": 0,
    "react/default-props-match-prop-types": 0,
    "react/no-danger": 0,
    "react/no-unescaped-entities": 0,
    "react/prop-types": 0,
    "react/require-default-props": 0,
    "react/sort-comp": 0
  },
  "env": {
    "jest": true,
    "jsx-control-statements/jsx-control-statements": true
  },
  "settings": {
    "import/resolver": {
      "babel-plugin-root-import": {
        "extensions": [".js", ".jsx", ".ts", ".tsx"]
      }
    }
  }
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```ts
declare var idx: number;

/* anything else */
```

```bash
eslint --ext .js,.jsx,.ts,.tsx web test
```

**What did you expect to happen?**
The `no-rule` rule should not detect Typescript's `declare var` as a variable declarations, as it is a type declaration and is stripped out when compiling.

**What actually happened? Please include the actual, raw output from ESLint.**
```
error  Unexpected var, use let or const instead                         no-var
```

**Are you willing to submit a pull request to fix this bug?**
Yes. It is as simple as replacing
```js
if (node.kind === "var") {
  report(node);
}
```
with
```js
if (node.kind === "var" && !node.declare) {
  report(node);
}
```
in `lib/rules/no-var.js`. It also affects the `vars-on-top` rule, which I'm not sure if should actually ignore `declare var` by default, but it would be helpful to allow it to be configured.

**Comment by g-plane** [comment]
Hi @vhfmag, thanks for the issue. You can use `declare let` for workaround.

**Comment by vhfmag** [comment]
Aaaand I learned something today. Thanks! Should I close the issue?

**Comment by g-plane** [comment]
Sure!
